packages/muelu/research/regionMG/src/SetupRegionHierarchy_def.hpp

```diff
@@ -736,31 +721,30 @@ void MakeInterfaceScalingFactors(const int maxRegPerProc,
 
 
 template<class Scalar, class LocalOrdinal, class GlobalOrdinal, class Node>
-void createRegionHierarchy(const int maxRegPerProc,
-                           const int numDimensions,
-                           const Array<Array<int> > lNodesPerDim,
-                           const Array<std::string> aggregationRegionType,
+void createRegionHierarchy(const int numDimensions,
+                           const Array<int> lNodesPerDim,
+                           const std::string aggregationRegionType,
                            RCP<Teuchos::ParameterList>& interfaceParams,
                            const std::string xmlFileName,
-                           Array<RCP<Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> > >& nullspace,
-                           Array<RCP<Xpetra::MultiVector<typename Teuchos::ScalarTraits<Scalar>::coordinateType, LocalOrdinal, GlobalOrdinal, Node> > >& coordinates,
-                           std::vector<RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> > >& regionGrpMats,
+                           RCP<Xpetra::MultiVector<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& nullspace,
+                           RCP<Xpetra::MultiVector<typename Teuchos::ScalarTraits<Scalar>::coordinateType, LocalOrdinal, GlobalOrdinal, Node> >& coordinates,
+                           RCP<Xpetra::Matrix<Scalar, LocalOrdinal, GlobalOrdinal, Node> >& regionGrpMats,
                            const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > mapComp,
-                           const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > rowMapPerGrp,
-                           const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > colMapPerGrp,
-                           const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > revisedRowMapPerGrp,
-                           const std::vector<RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > > revisedColMapPerGrp,
-                           const std::vector<RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> > > rowImportPerGrp,
+                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > rowMapPerGrp,
+                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > colMapPerGrp,
+                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > revisedRowMapPerGrp,
+                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > revisedColMapPerGrp,
+                           const RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> > rowImportPerGrp,
```

Comment:
The removed `std::vector` was related to the concept of "groups". Hence, this could be renamed to 
```suggestion
                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > rowMap,
                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > colMap,
                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > revisedRowMap,
                           const RCP<Xpetra::Map<LocalOrdinal, GlobalOrdinal, Node> > revisedColMap,
                           const RCP<Xpetra::Import<LocalOrdinal, GlobalOrdinal, Node> > rowImport,
```